code/scripting/api/objs/shipclass.cpp

```diff
@@ -198,6 +198,97 @@ ADE_VIRTVAR(ManufacturerString, l_Shipclass, "string", "Ship class manufacturer"
 		return ade_set_args(L, "s", "");
 }
 
+ADE_VIRTVAR(LengthString, l_Shipclass, "string", "Ship class length", "string", "Length, or empty string if handle is invalid")
+{
+	int idx;
+	const char* s = nullptr;
+	if(!ade_get_args(L, "o|s", l_Shipclass.Get(&idx), &s))
+		return ade_set_error(L, "s", "");
+
+	if(idx < 0 || idx >= ship_info_size())
+		return ade_set_error(L, "s", "");
+
+	ship_info *sip = &Ship_info[idx];
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "Setting Length is not supported");
+	}
+
+	if(sip->manufacturer_str != NULL)
```

Comment:
why are you checking against manufacturer_str  and not ship_length.
Also, please make this NULL into a nullptr

Reply:
Copy Paste strikes again. Fixed